### ISSUE TITLE
each zoom action appears to be replayed on changing each frame

### ISSUE THREAD

**Issue by mr-martian** [SwatPhonLab/UltraTrace#64]
_Originally posted by @jonorthwash at 2019-06-25T17:00:30Z ([original issue](https://github.swarthmore.edu/PhonLab/dicom-annotator/issues/64))_

If you zoom a lot and then change frames, you get messages like this on each frame load:

```
<ButtonPress event state=Mod2 num=4 x=326 y=279>
HARGLE BARGLE
<ButtonPress event state=Mod2 num=4 x=326 y=279>
HARGLE BARGLE
<ButtonPress event state=Mod2 num=4 x=326 y=279>
HARGLE BARGLE
<ButtonPress event state=Mod2 num=5 x=222 y=245>
HARGLE BARGLE
<ButtonPress event state=Mod2 num=5 x=222 y=245>
HARGLE BARGLE
<ButtonPress event state=Mod2 num=4 x=222 y=245>
HARGLE BARGLE
<ButtonPress event state=Mod2 num=4 x=222 y=245>
HARGLE BARGLE
<ButtonPress event state=Mod2 num=5 x=303 y=221>
HARGLE BARGLE
<ButtonPress event state=Mod2 num=5 x=328 y=237>
HARGLE BARGLE
<ButtonPress event state=Mod2 num=5 x=328 y=237>
HARGLE BARGLE
<ButtonPress event state=Mod2 num=4 x=328 y=237>
HARGLE BARGLE
<ButtonPress event state=Mod2 num=4 x=328 y=237>
HARGLE BARGLE
<ButtonPress event state=Mod2 num=5 x=405 y=259>
HARGLE BARGLE
<ButtonPress event state=Mod2 num=5 x=404 y=259>
HARGLE BARGLE
<ButtonPress event state=Mod2 num=4 x=404 y=259>
HARGLE BARGLE
<ButtonPress event state=Mod2 num=4 x=404 y=259>
HARGLE BARGLE
<ButtonPress event state=Mod2 num=4 x=404 y=259>
HARGLE BARGLE
<ButtonPress event state=Mod2 num=4 x=404 y=259>
HARGLE BARGLE
<ButtonPress event state=Mod2 num=5 x=397 y=250>
HARGLE BARGLE
<ButtonPress event state=Mod2 num=5 x=396 y=250>
HARGLE BARGLE
<ButtonPress event state=Mod2 num=5 x=394 y=252>
HARGLE BARGLE
<ButtonPress event state=Mod2 num=5 x=394 y=253>
HARGLE BARGLE
<ButtonPress event state=Mod2 num=5 x=394 y=253>
HARGLE BARGLE
<ButtonPress event state=Mod2 num=4 x=394 y=253>
HARGLE BARGLE
<ButtonPress event state=Mod2 num=4 x=394 y=253>
HARGLE BARGLE
<ButtonPress event state=Mod2 num=4 x=394 y=253>
HARGLE BARGLE
<ButtonPress event state=Mod2 num=4 x=394 y=253>
HARGLE BARGLE
<ButtonPress event state=Mod2 num=4 x=394 y=253>
HARGLE BARGLE
<ButtonPress event state=Mod2 num=5 x=388 y=250>
HARGLE BARGLE
<ButtonPress event state=Mod2 num=5 x=386 y=250>
HARGLE BARGLE
```

This appears to be because it's replaying the zoom events.  (This doesn't seem to be the case with pan events?)

Really it should remember the view (zoom level and pan position) as a finite set of small values and reload that; otherwise if people play around with zoom it will get really unwieldy.

**Comment by mr-martian** [comment]
_Originally posted by @jonorthwash at 2019-06-25T17:00:48Z_

@mr-martian, this appears to be related to your code too.  Mind looking at it?

**Comment by mr-martian** [comment]
_Originally posted by @mr-martian at 2019-06-25T20:31:13Z_

Fixed in 663c12d1923dc8d2c9917e29226bdae18bea3e57
